lib/fathead/perl_doc/Parsers/parse.pl

```diff
@@ -0,0 +1,963 @@
+
+sub aliases_cli_switches {
+    my ($switch) = @_;
+    $switch =~ s/\s.+$//;
+    my @switches = $switch =~ /^-\[(\w+)\]$/
+        ? (map { "-$_" } split '', $1)
+        : ($switch);
+    return map { (
+        "$_",
+        "$_ option",
+        "$_ flag",
+        "$_ switch",
+    ) } @switches;
+}
+
+sub parse_cli_switches {
+    ul_list_parser(
+        selector_main => 'a[name="Command-Switches"]',
+        title => sub { $_[0]->find('b')->first->all_text },
+        aliases => sub { aliases_cli_switches($_[1]) },
+    )->(@_);
+}
+
+#######################################################################
+#                             Diagnostics                             #
+#######################################################################
+
+sub aliasas_diag_messages {
+    my ($message) = @_;
+    $message =~ s/%\w+//g;
+    return ($message);
+}
+
+sub parse_diag_messages {
+    ul_list_parser(
+        selector_main => 'a[name="DESCRIPTION"]',
+        title => sub { $_[0]->find('b')->first->text },
+        aliases => sub { aliasas_diag_messages($_[1]) },
+    )->(@_);
+}
+
+#######################################################################
+#                              Operators                              #
+#######################################################################
+
+my @op_types = ("Binary", "Ternary", "Unary");
+my $op_re = join '|', @op_types;
+
+sub aliases_operators {
+    my ($operator) = @_;
+    my ($op_type, $op_name) = $operator =~ /^($op_re) "([^"]+)"/;
+    return (
+        "$op_name", "$op_name operator",
+        "$op_type $op_name", "$op_type $op_name operator",
+    );
+}
+
+sub parse_operators {
+    my ($self, $dom) = @_;
+    my @operators = $dom->find('p')->grep(
+        sub { $_->text =~ /^($op_re)/ }
+    )->each;
+    my @articles;
+    my @aliases;
+    foreach my $op (@operators) {
+        my $text = $op->all_text;
+        my ($op_type, $op_name) = $text =~ /^($op_re) ([^\s]+)/;
+        my $title = "$op_type $op_name operator";
+        my $link = $op->preceding('a[name] ~ h2')
+            ->last->preceding('a[name]')->last->{name};
+        map { push @aliases, { new => $_, orig => $title } }
+            (aliases_operators($title));
+        push @articles, {
+            anchor => $link,
+            text   => $text,
+            title  => $title,
+        };
+    }
+    return {
+        articles => \@articles,
+        aliases  => \@aliases,
+    };
+}
+
+# For docs like perlfunc, perlvar with multiple headers per entry.
+
+sub parse_variables {
+    my ($self, $dom) = @_;
+    my @mod_sections = $dom->at('a[name="General-Variables"]')
+        ->following('ul')->each;
+    ul_list_parser(
+        uls => \@mod_sections,
+        title => sub { $_[0]->find('b')->first->text . ' (variable)' },
+        aliases => sub { $_[1] =~ s/ \(variable\)//r },
+        categories => sub { ['Variables'] },
```

Comment:
These need to be more specific as categories are created globally. i.e "Perl Variables"

Otherwise the search "variables" would show this category, which isn't expected